### DISCUSSION TITLE
Replace file-loader with asset/resource using publicPath

### DISCUSSION THREAD

**Discussion by ldrick** [webpack/webpack#14920] [Q&A]
Hi, I had in webpack 4 file-loader to put Fontawesome fonts into a specific nested directory, but telling via publicPath to use an fixed directory. This looked like that:
```javascript
{
  test: [/fontawesome-free.*\.(woff|woff2|eot|ttf|svg)$/],
  loader: 'file-loader',
  options: {
    name: 'my/nested/path/fontawesome/[name].[ext]',
    publicPath: (url, resourcePath) => {
      const filePath = path.resolve(resourcePath);
      return `fontawesome/${path.basename(filePath)}`;
    },
  },
},
```
Now since in webpack5 this is deprecated, I tried to switch to asset/resource, but could not make it work like this:
```javascript
{
  test: [/fontawesome-free.*\.(woff|woff2|eot|ttf|svg)$/],
  type: 'asset/resource',
  generator: {
    filename: 'my/nested/path/fontawesome/[name][ext]',
    publicPath: 'fontawesome/[name][ext]',
  },
},
```
The `filename` works as expected, but setting any `publicPath` except `undefined` throws exceptions. Unfortunately I couldn't find any documentation, how to get `publicPath` working to behave like file-loader previous. Could you please help me, thanks in advance.

**Comment by alexander-akait** [comment]
Try to avoid using `publicPath`, default value is `auto`, so you will have relative URLs by default

**Reply by ldrick** [reply]
Unfortunately not having a relative url is exactly what is needed from my project. So I do know the path, the files will be accessable later, even that they are built to an nested directory.

**Reply by alexander-akait** [reply]
Hm, can you provide example of the repo and describe - what you have and what you expected and I will show how to fix it

**Reply by ldrick** [reply]
Thank you for trying to understand my question. I prepared a repo, which is stripped down to the necessary things and it also shows my workaround I did today, just to get the rest of the project further.
https://github.com/ldrick/webpack_asset_resource
You can see where the files are builded to and once you build it, you will see in `dist/files/themes/mytheme/app.css` how paths to fontawesome are inside. And once you unquote my current workaround in webpack, you will see, what the paths inside app.css should look like: `fontawesome/[name][ext]`
Again, thank you very much

**Reply by alexander-akait** [reply]
I think you faced with https://github.com/webpack-contrib/mini-css-extract-plugin/issues/872, yep, it sounds like a bug, it is in my TODO, hope I will search bug or how we can deal with this on this week, sorry

**Reply by ldrick** [reply]
Thanks @alexander-akait  for pointing me there, that seems to be my issue, too. I will add myself to the watchlist.
